test/unit/extension_video_sensing.js

```diff
@@ -0,0 +1,408 @@
+
+const {PNG} = require('pngjs');
+const {test} = require('tap');
+
+const VideoSensing = require('../../src/extensions/scratch3_video_sensing/index.js');
+const VideoMotion = require('../../src/extensions/scratch3_video_sensing/library.js');
+
+/**
+ * Prefix to the mock frame images used to test the video sensing extension.
+ * @type {string}
+ */
+const pngPrefix = 'extension_video_sensing_';
+
+/**
+ * Map of frame keys to the image filenames appended to the pngPrefix.
+ * @type {object}
+ */
+const framesMap = {
+    center: 'center',
+    left: 'left-5',
+    left2: 'left-10',
+    down: 'down-10'
+};
+
+/**
+ * Asynchronously read a png file and copy its pixel data into a typed array
+ * VideoMotion will accept.
+ * @param {string} name - partial filename to read
+ * @returns {Promise.<Uint32Array>} pixel data of the image
+ */
+const readPNG = name => (
+    new Promise((resolve, reject) => {
+        const png = new PNG();
+        createReadStream(join(__dirname, `${pngPrefix}${name}.png`))
+            .pipe(png)
+            .on('parsed', () => {
+                // Copy the RGBA pixel values into a separate typed array and
+                // cast the array to Uint32, the array format VideoMotion takes.
+                resolve(new Uint32Array(new Uint8ClampedArray(png.data).buffer));
+            })
+            .on('error', reject);
+    })
+);
+
+/**
+ * Read all the frames for testing asynchrnously and produce an object with
+ * keys following the keys in framesMap.
+ * @returns {object} mapping of keys in framesMap to image data read from disk
+ */
+const readFrames = (() => {
+    // Use this immediately invoking function expression (IIFE) to delay reading
+    // once to the first test that calls readFrames.
+    let _promise = null;
+
+    return () => {
+        if (_promise === null) {
+            _promise = Promise.all(Object.keys(framesMap).map(key => readPNG(framesMap[key])))
+                .then(pngs => (
+                    Object.keys(framesMap).reduce((frames, key, i) => {
+                        frames[key] = pngs[i];
+                        return frames;
+                    }, {})
+                ));
+        }
+        return _promise;
+    };
+})();
+
+/**
+ * Match if actual is within optMargin to expect. If actual is under -180,
+ * match if actual + 360 is near expect. If actual is over 180, match if actual
+ * - 360 is near expect.
+ * @param {number} actual - actual angle in degrees
+ * @param {number} expect - expected angle in degrees
+ * @param {number} optMargin - allowed margin between actual and expect in degrees
+ * @returns {boolean} true if actual is close to expect
+ */
+const isNearAngle = (actual, expect, optMargin = 10) => (
+    ((actual < expect + optMargin) && (actual > expect - optMargin)) ||
+    ((actual + 360 < expect + optMargin) && (actual + 360 > expect - optMargin)) ||
+    ((actual - 360 < expect + optMargin) && (actual - 360 > expect - optMargin))
+);
```

Comment:
Super minor: the wrapping part of this might be cleaner to implement in terms of modulo (`%`), or in terms of `wrapClamp`(*) from `math-util.js`. I could also see `isNearAngle` moving into `math-util.js` in case anyone else wants to use it.

(*) That said, I was surprised to see how our current implementation of `wrapClamp` works just now -- I'm not confident that it's retaining as much precision as it could.